microsoft-edge/test-and-automation/test-and-automation.md

```diff
@@ -0,0 +1,86 @@
+| Playwright | The Playwright library provides cross-browser automation through a single API. |
+| Puppeteer | The Puppeteer library provides a high-level API to control Microsoft Edge using the DevTools Protocol. |
+| WebDriver | Automates testing in Microsoft Edge by simulating user interaction.  Provides advantages over JavaScript unit tests. |
+| webhint | Checks your code for best practices and common errors, to test and improve accessibility, performance, cross-browser and PWA compatibility, and security of your site. |
+
+These tools are described below.
+
+
+<!-- ====================================================================== -->
+## DevTools Protocol
+
+Use the DevTools Protocol to instrument, inspect, debug, and profile browsers including Microsoft Edge.  The Microsoft Edge DevTools Protocol matches the APIs of the Chrome DevTools Protocol.
+
+See [DevTools Protocol](devtools-protocol.md).
+
+
+<!-- ====================================================================== -->
+## Origin Trials
+
+You can use Origin Trials to try out experimental APIs on live sites for a limited period of time.  When using Origin Trials, users of Microsoft Edge that visit your site may run code that uses experimental APIs.  To access the experimental APIs on each user machine, you don't need to go to `edge://flags` and turn on feature flags.
+
+See [Use Origin Trials in Microsoft Edge](../origin-trials/index.md).
+
+
+<!-- ====================================================================== -->
+## Playwright
+
+The Playwright library provides cross-browser automation through a single API.  Playwright enables cross-browser web automation that is evergreen, capable, reliable, and fast.
+
+Playwright launches browsers as headless, by default.  Headless browsers don't display a UI; so generally you use the command line - however, you can also configure Playwright to run the full Microsoft Edge UI.
+
+See [Use Playwright to automate and test in Microsoft Edge](../playwright/index.md).
+
+
+<!-- ====================================================================== -->
+## Puppeteer
+
+The Puppeteer library provides a high-level API to control Microsoft Edge using the DevTools Protocol.
+
+Puppeteer launches headless browsers by default.  Headless browsers don't display a UI, so you must use the command line.  You can also configure Puppeteer to run full (non-headless) Microsoft Edge.
+
+With Microsoft Edge, you can use `puppeteer-core`, a lightweight version of Puppeteer that launches an existing browser installation, like Microsoft Edge.
+
+See [Puppeteer overview](../puppeteer/index.md).
+
+
+<!-- ====================================================================== -->
+## Use WebDriver to automate Microsoft Edge
+
+WebDriver allows you to automate Microsoft Edge by simulating user interaction.  WebDriver's testing and simulation has advantages over JavaScript unit tests:
```

Comment:
```suggestion
WebDriver allows you to automate Microsoft Edge by simulating user interaction.  Tests that use WebDriver have some advantages over JavaScript unit tests that run in the browser:
```

Reply:
Applied fix 2x:
here
top of WebDriver article